test/clojure-mode-util-test.el

```diff
@@ -136,7 +136,49 @@
       (expect (buffer-string) :to-equal
               "(ns my-app.core
     (:require [my-app.views [user-page :as user-page]]
-              [rum.core :as rum] ;comment\n))")))
+              [rum.core :as rum] ;comment
+))")))
+
+  (it "should sort requires in a basic ns with comments in the end"
+    (with-clojure-buffer "(ns my-app.core
+    (:require [rum.core :as rum] ;comment
+              [my-app.views [user-page :as user-page]]
+              ;;[comment2]
+))"
+      (clojure-sort-ns)
+      (expect (buffer-string) :to-equal
+              "(ns my-app.core
+    (:require [my-app.views [user-page :as user-page]]
+              [rum.core :as rum] ;comment
+
+              ;;[comment2]
+))")))
+  (it "should sort requires in ns with copyright disclamer and comments"
+   (with-clojure-buffer ";; Copyright (c) John Doe. All rights reserved.
+;; The use and distribution terms for this software are covered by the
+;; Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+(ns clojure.core
+  (:require
+   ;; The first comment
+   [foo] ;; foo comment
+   ;; Middle comment
+   [bar] ;; bar comment
+   ;; A last comment
+   ))"
+      (clojure-sort-ns)
+      (expect (buffer-string) :to-equal
+              ";; Copyright (c) John Doe. All rights reserved.
+;; The use and distribution terms for this software are covered by the
+;; Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+(ns clojure.core
+  (:require
+   ;; Middle comment
+   [bar] ;; bar comment
+   ;; The first comment
+   [foo] ;; foo comment
+
+   ;; A last comment
```

Comment:
I haven't understood you completely.
Is it about additional linebreak or the sorting order itself?
The linebreak issue is not exactly related to this function.

Reply:
In either case it will be better to handle it in a separate PR.
Will be happy to discuss it with you further on slack.

I have merged the master and added a Changelog entry.Please feel free to squash the commits as you see it.

Reply:
Ah. But that is the expected behaviour currently.
`Comments at the start of a line are considered part of the following sexp.`
The sorting happens only based on what is inside the import. The comments are moved along with the sexp.

Reply:
Got it. 

IMHO the comments should be moved along with the imports instead of being sorted separately. So I feel the current behaviour makes some sense. Whether the comments should be placed below or above the sexp is up for debate.

Also if we make any changes to the sorting order it will break the current user experience. So a discussion + new ticket is necessary.

Reply:
Thanks again for the review and the merge. :)